hazelcast/src/main/java/com/hazelcast/nio/tcp/AbstractIOSelector.java

```diff
@@ -102,7 +102,16 @@ private void processSelectionQueue() {
             if (runnable == null) {
                 return;
             }
-            runnable.run();
+            if (runnable instanceof WriteHandler) {
```

Comment:
What I often do, and it certainly is not a hard rule, and perhaps it might not be needed here, is that I pull  the processing of a task out of the loop into a separate method.